network/identity.go

```diff
@@ -26,36 +26,48 @@ var (
 type identity struct {
 	proto.UnimplementedIdentityServer
 
-	pending     sync.Map
-	pendingSize int64
+	pending sync.Map
+
+	pendingInboundCount int64
+
+	pendingOutboundCount int64
 
 	srv *Server
 
 	initialized uint32
 }
 
-func (i *identity) numPending() int64 {
-	return atomic.LoadInt64(&i.pendingSize)
+func (i *identity) pendingInboundConns() int64 {
+	return atomic.LoadInt64(&i.pendingInboundCount)
+}
+
+func (i *identity) pendingOutboundConns() int64 {
+	return atomic.LoadInt64(&i.pendingOutboundCount)
 }
 
 func (i *identity) isPending(id peer.ID) bool {
-	val, ok := i.pending.Load(id)
-	if !ok {
-		return false
-	}
+	_, ok := i.pending.Load(id)
 
-	return val.(bool)
+	return ok
 }
 
 func (i *identity) delPending(id peer.ID) {
-	if _, loaded := i.pending.LoadAndDelete(id); loaded {
-		atomic.AddInt64(&i.pendingSize, -1)
+	if value, loaded := i.pending.LoadAndDelete(id); loaded {
+		if value.(network.Direction) == network.DirInbound {
+			atomic.AddInt64(&i.pendingInboundCount, -1)
+		} else if value.(network.Direction) == network.DirOutbound {
+			atomic.AddInt64(&i.pendingOutboundCount, -1)
+		}
 	}
 }
 
-func (i *identity) setPending(id peer.ID) {
-	if _, loaded := i.pending.LoadOrStore(id, true); !loaded {
-		atomic.AddInt64(&i.pendingSize, 1)
+func (i *identity) setPending(id peer.ID, direction network.Direction) {
+	if _, loaded := i.pending.LoadOrStore(id, direction); !loaded {
+		if direction == network.DirInbound {
+			atomic.AddInt64(&i.pendingInboundCount, 1)
+		} else if direction == network.DirOutbound {
+			atomic.AddInt64(&i.pendingOutboundCount, 1)
+		}
```

Comment:
Same here, please break it up into a switch statement for readability

Reply:
If anybody reading this, this is the commit that fixes it:

https://github.com/0xPolygon/polygon-sdk/pull/326/commits/7fa13584e2cdc484c1189865c7b2a51bc7ecec5c